src/encoded/static/components/search.js

```diff
@@ -157,6 +157,40 @@ PickerActions.contextTypes = {
 };
 
 
+/**
+ * Displays the search result data line for target objects when the parent object's targets
+ * comprise an array. The parent object's schema must include the target array's @id in its
+ * `columns` property.
+ */
+export const TargetsDataLine = ({ result, targetPropertyName }) => {
+    if (result[targetPropertyName] && result[targetPropertyName].length > 0) {
+        return (
+            <>
+                <span className="result-item__property-title">Target{result[targetPropertyName].length > 1 ? 's' : ''}: </span>
+                {result[targetPropertyName].map((target, i) => (
+                    <React.Fragment key={target['@id']}>
+                        {i > 0 ? ', ' : null}
+                        <a href={target['@id']} aria-label={`Target ${target.label}`}>{target.label}</a>
+                    </React.Fragment>
+                ))}
+            </>
+        );
+    }
+    return null;
+};
```

Comment:
It's too bad you can't use .join for cases like this. I guess you could also use .reduce. But this is a good solution, very readable.

Reply:
Yeah, there’s another [solution](https://stackoverflow.com/questions/34034038/how-to-render-react-components-by-using-map-and-join/35840806#answer-40788571) that I think Phil found, but I didn’t see any advantage over this method, and it was more difficult to understand, I thought.

Reply:
I actually found that exact link when I was going down the rabbit hole of why can't you use .join!